specifyweb/stored_queries/group_concat.py

```diff
@@ -25,3 +28,35 @@ def process_clause(idx):
 
     return 'GROUP_CONCAT(%s)' % inner
 
+def group_by_displayed_fields(query: QueryConstruct, fields):
+    # fields = extract_fields_and_aliases(query.query) TODO: remove this line
+    for field in fields:
+        query = query.group_by(field)
+    
+    return query
+
+# TODO: remove this function
+def extract_fields_and_aliases(query):
+    query_str = str(query.statement.compile(compile_kwargs={"literal_binds": True}))
+    
+    select_part_match = re.search(r"SELECT(.*?)FROM", query_str, re.IGNORECASE | re.DOTALL)
+    if not select_part_match:
+        return []
+    
+    select_part = select_part_match.group(1)
+    
+    # Match each field expression in the SELECT clause
+    field_pattern = r"(.*?)(?: AS ([\w]+))?(,|$)"
+    
+    matches = re.findall(field_pattern, select_part, re.IGNORECASE | re.DOTALL)
+
+    fields_and_aliases = []
+    for match in matches:
+        field_expr, alias, _ = match
+        field_expr = field_expr.strip()
+        if alias:
+            fields_and_aliases.append(alias)
+        elif field_expr:
+            fields_and_aliases.append(field_expr)
+
+    return fields_and_aliases
```

Comment:
Bad, get rid of this. Why not rely more on SQLAlchemy?

Reply:
Ya, I'm definitely getting rid of this.  I was looking in out SQLAlchemy code for where we set the alias name of a field, so I could just put the field's alias name in the group by statement instead of the full expression.